alpha/alpha-core/src/main/java/org/apache/servicecomb/saga/alpha/core/CompositeOmegaCallbackRunner.java

```diff
@@ -20,30 +20,44 @@
 import static java.util.Collections.emptyMap;
 
 import java.lang.invoke.MethodHandles;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.Callable;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class CompositeOmegaCallback implements OmegaCallback {
+public class CompositeOmegaCallbackRunner implements OmegaCallback, Callable<List<TxEvent>> {
```

Comment:
Compensate Thread ， using by PushBackOmegaCallback . PushBackOmegaCallback  will filter all compensate event by there service name ,do all compensate event list with the same service name  in one  CompositeOmegaCallbackRunner  thread and only return those  compensate events which are successful.

Reply:
since omega call back is not thread safe. so compensate the same service use one thread.use other thread to compensate other service.